sources/Interop/D3D12MemoryAllocator/src/ALLOCATION_DESC.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright © Tanner Gooding and Contributors. Licensed under the MIT License (MIT). See License.md in the repository root for more information.
+
+namespace TerraFX.Interop
+{
+    /// <summary>Parameters of created D3D12MA::Allocation object. To be used with Allocator::CreateResource.</summary>
+    public unsafe struct ALLOCATION_DESC
+    {
+        /// <summary>Flags.</summary>
+        public ALLOCATION_FLAGS Flags;
+
+        /// <summary>
+        /// The type of memory heap where the new allocation should be placed.
+        /// <para>It must be one of: `D3D12_HEAP_TYPE_DEFAULT`, `D3D12_HEAP_TYPE_UPLOAD`, `D3D12_HEAP_TYPE_READBACK`.</para>
+        /// <para>When D3D12MA::ALLOCATION_DESC::CustomPool != NULL this member is ignored.</para>
+        /// </summary>
```

Comment:
```suggestion
        /// <summary>The type of memory heap where the new allocation should be placed.</summary>
        /// <remarks>
        ///     <para>It must be one of: `D3D12_HEAP_TYPE_DEFAULT`, `D3D12_HEAP_TYPE_UPLOAD`, `D3D12_HEAP_TYPE_READBACK`.</para>
        ///     <para>When D3D12MA::ALLOCATION_DESC::CustomPool != NULL this member is ignored.</para>
        /// </remarks>
```

You could optionally fix the relevant bits to be proper `<see cref="D3D12_HEAP_TYPE_DEFAULT" />` entries as well
Likewise, you could C#-ify the `D3D12MA::ALLOCATION_DESC::CustomPool != NULL` code to be `<c><see cref="CustomPool" /> != null</c>`

Reply:
I've found it generally good to keep the summary short and sweet, one not long sentence. Additional info can be placed in `<remarks>`. 

This is because IntelliSense generally has a fairly limited amount of space to show you info, while the proper docs website has a lot more room.

Reply:
The same applies for most of the doc comments. I don't mind so much on the `cref` or `C#-ification` of commented bits in the initial PR. We can log an issue for someone to pick up later since it is a lot more work.

Reply:
Done in b392084c646cf9f674dff8a2b589e2e9aab099af 👍